pkgs/tools/networking/smartdns/default.nix

```diff
@@ -0,0 +1,54 @@
+
+{ stdenv, fetchFromGitHub, openssl, fetchpatch }:
+
+stdenv.mkDerivation rec {
+  pname = "smartdns";
+  version = "29";
+  src = fetchFromGitHub {
+    owner = "pymumu";
+    repo = pname;
+    rev = "Release${version}";
+    sha256 = "06kgmahgma0ybhacqzj5sfazmhzs3p27a0sxik4xbbk3dyqn9ykj";
+  };
+
+  buildInputs = [ openssl ];
+
+  patches = [
+    (fetchpatch {
+      url =
+        "https://git.archlinux.org/svntogit/community.git/plain/trunk/systemd.patch?h=packages/smartdns";
```

Comment:
This URL is not stable (it is obtained from `HEAD` of the `community.git` repository), you need to pin it to a certain commit e.g. `https://git.archlinux.org/svntogit/community.git/plain/trunk/systemd.patch?h=packages/smartdns&id=c15e81742c93e24c47de59a956f1d0d0f7156f0a`.

Reply:
I will re-push it again

Reply:
Could you also report the issue about `PIDFile` using outdated directory upstream (see FHS 3.0 notes for it https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch05s13.html).

Reply:
> Could you also report the issue about `PIDFile` using outdated directory upstream (see FHS 3.0 notes for it https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch05s13.html).

Yes, that's the case. But by the time, should I proceed with the patch?

Reply:
And also about `StartLimitIntervalSec` belonging to [`Unit` section](https://www.freedesktop.org/software/systemd/man/systemd.unit.html#StartLimitIntervalSec=interval), not [`Service` section](https://www.freedesktop.org/software/systemd/man/systemd.service.html).

Reply:
> And also about `StartLimitBurst` and `StartLimitIntervalSec` belonging to [`Unit` section](https://www.freedesktop.org/software/systemd/man/systemd.unit.html#StartLimitIntervalSec=interval), not [`Service` section](https://www.freedesktop.org/software/systemd/man/systemd.service.html).

It has been addressed in the patch from Arch Linux. Anyway, I will report.

Reply:
Yes, using the patch for now is okay but it should be eventually fixed upstream. For that reason either a TODO comment in the source code or, better, upstream issue should be created.

Reply:
https://github.com/pymumu/smartdns/issues/390
reported